src/clr-angular/forms/input/input-container.ts

```diff
@@ -23,19 +28,42 @@ import { NgControlService } from '../common/providers/ng-control.service';
             <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
         </div>
     `,
-  host: { '[class.clr-form-control]': 'true' },
-  providers: [IfErrorService, NgControlService],
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-row]': '!isVertical()',
+  },
+  providers: [IfErrorService, NgControlService, ControlIdService],
 })
-export class ClrInputContainer implements OnDestroy {
+export class ClrInputContainer implements DynamicWrapper, OnDestroy {
   subscription: Subscription;
   invalid = false;
+  _dynamic = false;
+  @ContentChild(ClrLabel) label: ClrLabel;
 
-  constructor(private ifErrorService: IfErrorService) {
+  constructor(private ifErrorService: IfErrorService, @Optional() private layoutService: LayoutService) {
```

Comment:
Should this `LayoutService` be optional here? In case it's missing, can we throw an error that the component should be placed inside a clrForm.

Reply:
It is possible to have a form control without the `clrForm`, so it is optional.